interfaces/repo.go

```diff
@@ -747,35 +747,30 @@ func (r *Repository) RemoveSnap(snapName string) error {
 
 // DisconnectSnap disconnects all the connections to and from a given snap.
 //
-// The return value is a list of names that were affected.
-func (r *Repository) DisconnectSnap(snapName string) ([]string, error) {
+// The return value is a set of snaps that were affected.
+func (r *Repository) DisconnectSnap(snapName string) (map[*snap.Info]bool, error) {
```

Comment:
So it now returns a set, so that you can add more affected snaps later (after plug-slot are connected, query connected slots and their corresponding snap infos). That way we avoid duplicates and don't have to merge/dedup if it was just a slice